### PR TITLE
Add az-formdata rule to suggest check on formData parameters

### PR DESCRIPTION
**Pull request by mikekistler**
Another simple rule, level "info", to flag formData parameters to check for proper use.

### CODE REVIEW

File: docs/azure-ruleset.md [hunk 1]
```diff
@@ -48,6 +48,14 @@ Every operation should have a default response with error response body.
 
 All `4xx` and `5xx` responses should specify `x-ms-error-response: true` except for `404` response of HEAD operation.
 
+### az-formdata
+
+Check for appropriate use of formData parameters.
+
+It can be appropriate to use formData parameters when sending multiple file-type parameters or an array of file-type parameters.
+But it is unnecessary and overly complicated to use formData to send a single file-type parameter.
```
Comment (markweitzel) [inline docs/azure-ruleset.md:56]:
```suggestion
But it is usually unnecessary and can be overly complicated to use formData when all you are doing is sending a single file-type parameter. 
```

File: docs/azure-ruleset.md [hunk 2]
```diff
@@ -48,6 +48,14 @@ Every operation should have a default response with error response body.
...
+This should be defined as a `body` parameter with `type: string, format: binary` and use content-type `application/octet-stream`.
```
Comment (markweitzel) [inline docs/azure-ruleset.md:57]:
```suggestion
Instead, consider defining a `body` parameter with `type: string, format: binary` and use content-type `application/octet-stream`.
```

**Review by markweitzel — APPROVED** [review]
Overall looks good. Because this is `info` I softened the language a bit--up to you if you want to take the suggestions. No worries if not.

**Comment by mikekistler** [comment]
Sorry I missed your suggested changes before hitting the merge button.  I'll incorporate these into my next PR.

**Pull request merged**